synapse/rest/client/v1/room.py

```diff
@@ -230,6 +233,14 @@ async def on_POST(self, request, room_id, event_type, txn_id=None):
             "sender": requester.user.to_string(),
         }
 
+        inherit_depth = False
+        prev_events = parse_strings_from_args(request.args, "prev_event")
+        if self._msc2716_enabled:
+            if prev_events and requester.app_service:
```

Comment:
Add check for `An event must not have more than 20 prev_events.` (from spec)

Reply:
I don't think we need to add an extra check because there is already a check for max 10 `prev_events` in [`create_new_client_event`](https://github.com/matrix-org/synapse/blob/8f75bf1df7f2bcb3ffe0bb89f8fe3351a48769c0/synapse/handlers/message.py#L796)